common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -1128,14 +1128,36 @@ def get_parent_location(self, locator, **kwargs):
 
         :param locator: BlockUsageLocator restricting search scope
         """
+        def is_root_block(xblock, is_root=False):
```

Comment:
This function should really be called `has_path_to_root`.

I like the idea here, that `get_parent_location` will only return parents in the course tree. This may help prevent many other orphan-related bugs! However, I also worry about if orphan's children ever need this information? I can't think of a scenario off the top of my head where they would need it.

@cpennington , I know you wanted to see this work happening in [`find_path_to_course`](https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/modulestore/search.py#L41). What do you think of instead putting it here?